scandium-core/src/main/java/org/eclipse/californium/scandium/util/PskUtil.java

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+ * Copyright 2018 University of Rostock, Institute of Applied Microelectronics and Computer Engineering
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Vikram (University of Rostock)- Initial creation, adapted from ClientHandshaker
+ ******************************************************************************/
+package org.eclipse.californium.scandium.util;
+
+import org.eclipse.californium.elements.auth.PreSharedKeyIdentity;
+import org.eclipse.californium.scandium.dtls.AlertMessage;
+import org.eclipse.californium.scandium.dtls.AlertMessage.AlertDescription;
+import org.eclipse.californium.scandium.dtls.AlertMessage.AlertLevel;
+import org.eclipse.californium.scandium.dtls.DTLSSession;
+import org.eclipse.californium.scandium.dtls.HandshakeException;
+import org.eclipse.californium.scandium.dtls.pskstore.PskStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Called from ClientHandshaker class to retrieve preshared key identity 
+ * and preshared key in PSK and ECDH_PSK cases.
+ *
+ */
+public class PskUtil {
+	
+	private static final Logger LOGGER = LoggerFactory.getLogger(PskUtil.class.getName());
+	
+	private byte[] psk;
+	
+	private PreSharedKeyIdentity pskIdentity;
+	
+	/**
+	 * Support Server Name Indication TLS extension.
+	 */
+	protected boolean sniEnabled = true;
+	
+	/**
+	 * Retrieves preshared key identity and preshared key for the given dtls session and psk store.
+	 * @param session - {@code DTLSSession}
+	 * @param pskStore - {@code PskStore}
+	 * @throws HandshakeException
+	 * @throws NullPointerException if either session or pskStore is {@code null}
+	 */
+	public PskUtil(DTLSSession session, PskStore pskStore) throws HandshakeException {
+		if(session == null) {
+			throw new NullPointerException("Dtls session cannot be null");
+		}
+		if(pskStore == null) {
+			throw new NullPointerException("psk store cannot be null");
+		}
+		configurePskCredentials(session, pskStore);
```

Comment:
I would just delete line 60-63, I don't see the benefit of split the implementation.